### PR TITLE
OSDOCS3082: Networking operators overview

### PR DESCRIPTION
**Pull request by xenolinux**
OCP version: 4.10+
There is a different [PR](https://github.com/openshift/openshift-docs/pull/45515) for 4.6+ version.

[Preview](https://deploy-preview-45517--osdocs.netlify.app/openshift-enterprise/latest/networking/networking-operators-overview.html)
[Jira](https://issues.redhat.com/browse/OSDOCS-3082)
QE contact @zhaozhanqi

### CODE REVIEW

**Comment by zhaozhanqi** [comment]
CNO part lgtm

cc @lihongan  to take a look at ingress and dns operator

**Review by pneedle-rh — COMMENTED** [review]
@xenolinux I have now completed my review of this PR. Please let me know if you have any questions about my suggestions. Thanks!

File: networking/networking-operators-overview.adoc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+include::_attributes/common-attributes.adoc[]
+:context: networking-operators-overview
+
+toc::[]
+
+{product-title} supports multiple types of networking Operators. You can manage the {product-title} cluster networking using these networking Operators.
+
+[id="networking-operators-overview-cluster-network-operator"]
+== Cluster Network Operator
+The Cluster Network Operator (CNO) deploys and manages the cluster network components on an OpenShift Container Platform cluster, including the Container Network Interface (CNI) default network provider plug-in selected for the cluster during installation. For more information, see xref:../networking/cluster-network-operator.adoc#cluster-network-operator[Cluster Network Operator in {product-title}].
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply networking/networking-operators-overview.adoc:13]: Ack'ed

[the remark this answers is not in the document]
Comment (xenolinux) [reply networking/networking-operators-overview.adoc:13]: Ack'ed

File: networking/networking-operators-overview.adoc [hunk 2]
```diff
@@ -0,0 +1,25 @@
+:_content-type: ASSEMBLY
+[id="networking-operators-overview"]
+= Networking Operators overview
+include::_attributes/common-attributes.adoc[]
+:context: networking-operators-overview
+
+toc::[]
+
+{product-title} supports multiple types of networking Operators. You can manage the {product-title} cluster networking using these networking Operators.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply networking/networking-operators-overview.adoc:9]: Ack'ed

File: networking/networking-operators-overview.adoc [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+[id="networking-operators-overview-cluster-network-operator"]
+== Cluster Network Operator
+The Cluster Network Operator (CNO) deploys and manages the cluster network components on an OpenShift Container Platform cluster, including the Container Network Interface (CNI) default network provider plug-in selected for the cluster during installation. For more information, see xref:../networking/cluster-network-operator.adoc#cluster-network-operator[Cluster Network Operator in {product-title}].
+
+[id="networking-operators-overview-dns-operator"]
+== DNS Operator
+The DNS Operator deploys and manages CoreDNS to provide a name resolution service to pods, enabling DNS-based Kubernetes Service discovery in OpenShift. xref:../networking/dns-operator.adoc#dns-operator[DNS Operator in {product-title}].
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply networking/networking-operators-overview.adoc:17]: Ack'ed

[the remark this answers is not in the document]
Comment (xenolinux) [reply networking/networking-operators-overview.adoc:17]: Ack'ed

File: networking/networking-operators-overview.adoc [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+[id="networking-operators-overview-ingress-operator"]
+== Ingress Operator
+When you create your OpenShift Container Platform cluster, pods and services running on the cluster are each allocated their own IP addresses. The IP addresses are accessible to other pods and services running nearby but are not accessible to outside clients. The Ingress Operator implements the IngressController API and is the component responsible for enabling external access to OpenShift Container Platform cluster services. For more information, see xref:../networking/ingress-operator.adoc#configuring-ingress[Ingress Operator in {product-title}].
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply networking/networking-operators-overview.adoc:21]: Ack'ed
Changed "outside clients" to "external clients". [ Yes, it refers to clients that are external to the cluster.]

File: networking/networking-operators-overview.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (xenolinux) [reply networking/networking-operators-overview.adoc:17]: Ack'ed

File: networking/networking-operators-overview.adoc [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+[id="networking-operators-overview-external-dns-operator"]
+== External DNS Operator
+The ExternalDNS Operator deploys and manages ExternaDNS to provide the name resolution service to pods, routes from the external DNS provider to OpenShift.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply networking/networking-operators-overview.adoc:25]: Ack'ed

[the remark this answers is not in the document]
Comment (xenolinux) [reply networking/networking-operators-overview.adoc:25]: Ack'ed

[the remark this answers is not in the document]
Comment (xenolinux) [reply networking/networking-operators-overview.adoc]: Ack'ed

[the remark this answers is not in the document]
Comment (xenolinux) [reply networking/networking-operators-overview.adoc]: Confirmed with the SME, the correct sentence would be :

The External DNS Operator deploys and manages ExternalDNS to provide the name resolution for services and routes from the external DNS provider to {product-title}.

File: networking/networking-operators-overview.adoc [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+{product-title} supports multiple types of networking Operators. You can manage the cluster networking using these networking Operators.
+
+[id="networking-operators-overview-cluster-network-operator"]
+== Cluster Network Operator
+The Cluster Network Operator (CNO) deploys and manages the cluster network components in an {product-title} cluster. This includes deployment of the Container Network Interface (CNI) default network provider plug-in selected for the cluster during installation. For more information, see xref:../networking/cluster-network-operator.adoc#cluster-network-operator[Cluster Network Operator in {product-title}].
+
+[id="networking-operators-overview-dns-operator"]
+== DNS Operator
+The DNS Operator deploys and manages CoreDNS to provide a name resolution service to pods. This enables DNS-based Kubernetes Service discovery in {product-title}. For more information, xref:../networking/dns-operator.adoc#dns-operator[DNS Operator in {product-title}].
```
Comment (Amrita42) [inline networking/networking-operators-overview.adoc:17]: I would suggest add , **see**  to keep it consistent with the rest of the xrefs here.

File: networking/networking-operators-overview.adoc [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+[id="networking-operators-overview-ingress-operator"]
+== Ingress Operator
+When you create your {product-title} cluster, pods and services running on the cluster are each allocated IP addresses. The IP addresses are accessible to other pods and services running nearby but are not accessible to external clients. The Ingress Operator implements the Ingress Controller API and is responsible for enabling external access to {product-title} cluster services. For more information, see xref:../networking/ingress-operator.adoc#configuring-ingress[Ingress Operator in {product-title}].
+
+[id="networking-operators-overview-external-dns-operator"]
+== External DNS Operator
+The External DNS Operator deploys and manages ExternalDNS to provide the name resolution for services and routes from the external DNS provider to {product-title}. For more information, xref:../networking/external_dns_operator/understanding-external-dns-operator.adoc#external-dns-operator[Understanding the External DNS Operator].
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply networking/networking-operators-overview.adoc]: Addressed

**Comment by mburke5678** [comment]
Closed in favor or https://github.com/openshift/openshift-docs/pull/47581

**Pull request closed without merging**